R/LearnerClassifXgboost.R

```
@@ -244,13 +245,22 @@ LearnerClassifXgboost = R6Class("LearnerClassifXgboost",
       )
 
       data = task$data(cols = task$feature_names)
-      # recode to 0:1 to that for the binary case the positive class translates to 1 (#32)
+      # recode to 0:1 so that for the binary case the positive class translates to 1 (#32)
       # note that task$truth() is guaranteed to have the factor levels in the right order
       label = nlvls - as.integer(task$truth())
-      data = xgboost::xgb.DMatrix(data = as_numeric_matrix(data), label = label)
+      xgb_data = xgboost::xgb.DMatrix(data = as_numeric_matrix(data), label = label)
 
       if ("weights" %in% task$properties) {
-        xgboost::setinfo(data, "weight", task$weights$weight)
+        xgboost::setinfo(xgb_data, "weight", task$weights$weight)
+      }
+
+      bm = pv$base_margin
+      pv$base_margin = NULL # silence xgb.train message
+      bm_is_feature = !is.null(bm) && is.character(bm) && (bm %in% task$feature_names)
+      # works only with binary classification objectives
+      obj_is_binary = startsWith(pv$objective, "binary")
```

otherwise it is just ignored? I think we should rather throw an error here.

---

yes